pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -650,124 +650,125 @@ func TestPullSourcesFromReference(t *testing.T) {
+			[]bool{false, true, false},
+		},
+		// Registry B allowing digests pull only from mirrors
+		{
+			"registry-b.com/foo",
+			"image",
+			[]string{"mirror-1.registry-b.com", "mirror-2.registry-b.com", "registry-b.com/bar"},
+			[]bool{false, false, false},
+			[]string{"registry-b.com/bar"},
+			[]bool{false},
+		},
 		// Registry A has mirrors allow any kind of pull
 		{
-			"registry-a.com/foo/image",
+			"registry-a.com/baz",
+			"image",
 			[]string{"mirror-1.registry-a.com", "mirror-2.registry-a.com", "registry-a.com/bar"},
+			[]bool{false, true, false},
 			[]string{"mirror-1.registry-a.com", "mirror-2.registry-a.com", "registry-a.com/bar"},
+			[]bool{false, true, false},
 		},
 		// Registry B has mirrors allow digests pull only
 		{
-			"registry-b.com/foo/image",
+			"registry-b.com/baz",
+			"image",
 			[]string{"mirror-1.registry-b.com", "mirror-2.registry-b.com", "registry-b.com/bar"},
+			[]bool{false, false, false},
 			[]string{"registry-b.com/bar"},
+			[]bool{false},
 		},
 		// Registry C has a mirror allows digest pull only and a mirror allows any kind of pull
 		{
-			"registry-c.com/foo/image",
+			"registry-c.com/baz",
+			"image",
 			[]string{"mirror-1.registry-c.com", "mirror-2.registry-c.com", "registry-c.com/bar"},
+			[]bool{false, false, false},
 			[]string{"mirror-1.registry-c.com", "registry-c.com/bar"},
+			[]bool{false, false},
 		},
 		// Registry D set digest-only for registry level, allows only digest pulls
 		// Registry D has no digest-only set for mirrors table
 		{
-			"registry-d.com/foo/image",
+			"registry-d.com/baz",
+			"image",
 			[]string{"mirror-1.registry-d.com", "mirror-2.registry-d.com", "registry-d.com/bar"},
+			[]bool{false, false, false},
 			[]string{"registry-d.com/bar"},
+			[]bool{false},
 		},
 		// Registry E has mirrors only allows tag pull
 		{
-			"registry-e.com/foo/image",
+			"registry-e.com/baz",
+			"image",
 			[]string{"registry-e.com/bar"},
+			[]bool{false},
 			[]string{"mirror-1.registry-e.com", "mirror-2.registry-e.com", "registry-e.com/bar"},
+			[]bool{false, false, false},
 		},
 		// Registry F has one tag only mirror does not allow digest pull
 		{
-			"registry-f.com/foo/image",
+			"registry-f.com/baz",
+			"image",
 			[]string{"mirror-1.registry-f.com", "registry-f.com/bar"},
+			[]bool{false, false},
 			[]string{"mirror-1.registry-f.com", "mirror-2.registry-f.com", "registry-f.com/bar"},
+			[]bool{false, false, false},
 		},
 		// Registry G has one digest-only pull and one tag only pull
 		{
-			"registry-g.com/foo/image",
+			"registry-g.com/baz",
+			"image",
 			[]string{"mirror-1.registry-g.com", "mirror-3.registry-g.com", "mirror-4.registry-g.com", "registry-g.com/bar"},
+			[]bool{false, false, false, false},
 			[]string{"mirror-2.registry-g.com", "mirror-3.registry-g.com", "mirror-4.registry-g.com", "registry-g.com/bar"},
+			[]bool{false, false, false, false},
 		},
 	} {
```

Comment:
Throughout, `filepath.Join` is a bad fit:
- It uses `os.PathSeparator` instead of `/`
- `filepath.Join` calls `filepath.Clean`. That doesn’t quite _hurt_ but it is not semantically correct.

Maybe hard-coding `/` (with `+"/"+` or `Sprintf` to combine all the elements, I don’t care which one) would be simplest.